gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2441,8 +2697,18 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
             nodata.fVal = NC_FILL_FLOAT;
         else
             nodata.dfVal = NC_FILL_DOUBLE;
-        status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
-                            &nVarID);
+
+        if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation